File: frontend/src/types/produto.ts
export type Produto = {
  id_produto: string
  nome_produto: string
  categoria_produto: string
  preco: number | null
  peso_produto_gramas: number | null
  comprimento_centimetros: number | null
  altura_centimetros: number | null
  largura_centimetros: number | null
}

export type ProdutoListItem = {
  id_produto: string
  nome_produto: string
  categoria_produto: string
  preco: number
  nome_vendedor: string
  avaliacao_media: number
}

export type PaginatedResponse<T> = {
  items: T[]
  total: number
  skip: number
  limit: number
  pages: number
  current_page: number
  has_next: boolean
  has_previous: boolean
}

File: backend/app/schemas/produto.py
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ProdutoBase(BaseModel):
    nome_produto: str
    categoria_produto: str
    imagem_url: Optional[str] = None
    preco: Optional[float] = None
    peso_produto_gramas: Optional[float] = None
    comprimento_centimetros: Optional[float] = None
    altura_centimetros: Optional[float] = None
    largura_centimetros: Optional[float] = None


class ProdutoCreate(ProdutoBase):
    pass


class ProdutoUpdate(BaseModel):
    nome_produto: Optional[str] = None
    categoria_produto: Optional[str] = None
    imagem_url: Optional[str] = None
    preco: Optional[float] = None
    peso_produto_gramas: Optional[float] = None
    comprimento_centimetros: Optional[float] = None
    altura_centimetros: Optional[float] = None
    largura_centimetros: Optional[float] = None


class ProdutoResponse(ProdutoBase):
    id_produto: str

    model_config = ConfigDict(from_attributes=True)


class ProdutoListItemResponse(BaseModel):
    id_produto: str
    nome_produto: str
    categoria_produto: str
    imagem_url: Optional[str] = None
    preco: float
    nome_vendedor: str
    avaliacao_media: float
